src/LaunchDarkly.EventSource/EventSource.cs

```diff
@@ -187,8 +187,6 @@ public async Task StartAsync()
                     // If the user called Close(), ReadyState = Shutdown, so errors are irrelevant.
                     if (ReadyState != ReadyState.Shutdown)
                     {
-                        Close(ReadyState.Closed);
-

```

Comment:
Changed this so the transition to the closed state happens after it's called OnError. May not matter for real applications, but it makes more sense in the tests to see (error, closed) rather than (closed, error).